### PR TITLE
Fixing error with PipelineOptions DisplayData of lists

### PR DESCRIPTION
**Pull request by pabloem**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by pabloem** [comment]
r: aaltay

File: sdks/python/apache_beam/internal/apiclient.py [hunk 1]
```diff
@@ -242,12 +243,28 @@ def __init__(self, packages, options, environment_version):
...
 
-      dd = DisplayData.create_from(options)
-      items = [item.get_dict() for item in dd.items]
+      items = self.format_options_display_data(options)
       self.proto.sdkPipelineOptions.additionalProperties.append(
           dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
               key='display_data', value=to_json_value(items)))
 
+  @classmethod
+  def format_options_display_data(cls, options):
```
Comment (aaltay) [inline sdks/python/apache_beam/internal/apiclient.py]: Why don't you do this at DisplayDataItem.__init__? if the value is an instance of list, convert value first and set the type based on converted value.

File: sdks/python/apache_beam/utils/options.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def get_all_options(self, drop_default=False):
     Returns:
       Dictionary of all args and values.
     """
-    parser = argparse.ArgumentParser()
+    parser = argparse.ArgumentParser(conflict_handler='resolve')
```
Comment (aaltay) [inline sdks/python/apache_beam/utils/options.py]: what is the reason for this change?

**Comment by aaltay** [comment]
Thank you. LGTM.

**Comment by pabloem** [comment]
@bjchambers can you please merge? : )

**Comment by pabloem** [comment]
Reopened pull request to rerun tests.
